cheroot/workers/monitor.py

```diff
@@ -0,0 +1,135 @@
+"""Monitor for threadpool."""
+__metaclass__ = type
+
+import threading
+import time
+
+from dynpool import DynamicPoolResizer
+
+
+class BackgroundTask(threading.Thread):
+    """A subclass of threading.Thread whose run() method repeats.
+
+    Use this class for most repeating tasks. It uses time.sleep() to wait
+    for each interval, which isn't very responsive; that is, even if you call
+    self.cancel(), you'll have to wait until the sleep() call finishes before
+    the thread stops. To compensate, it defaults to being daemonic, which means
+    it won't delay stopping the whole process.
+
+    :param int interval: time interval between invocation of input function
+    :param function: callable that needs to be invoked
+    :param args: list of arguments to pass to the function
+    :param kwargs: dictionary of keyword args to pass to the function
+    """
+
+    def __init__(self, interval, function, name=None, args=None, kwargs=None):
```

Comment:
So I have two thoughts about this:
1) `function` arg is tightly bound to `args` and `kwargs`, so putting `name` in the middle seems counterintuitive from the API perspective. So maybe do something like this:
```suggestion
    def __init__(
        self, interval,
        function, args=None, kwargs=None,
        name=None,
    ):
```
2) On the second thought, maybe it's a better API if this would only accept a callable. This way, end-users would have to pass a `functools.partial()` instance or `lambda`, or just some wrapper. This may look cleaner.

Reply:
I don't see any place where any args/kwargs are passed — it doesn't seem like they are needed

Reply:
1. This class is derived from threading.Thread whose init looks like: 
```
class Thread(group=None, target=None, name=None, args=(), kwargs=None, daemon=None)
```
Hence the name between target and its args in this class too. 
2. as of now args and kwargs are not used, but given that this is a generic BT class and can be used for any func callable, isn't it better to have the args and kwargs support?